core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/collection_sequence_difference/CollectionAndSequenceDifferenceTest.kt

```diff
@@ -0,0 +1,91 @@
+package com.baeldung.collection_sequence_difference
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class CollectionAndSequenceDifferenceTest {
+
+    //eagerly and lazy evaluation
+    @Test
+    fun `collection eagerly iterate through all the items in map`() {
+        var mapIterations = 0
+        (1..100)
+                .map {
+                    mapIterations++
+                    it * it
+                }
+                .filter { it % 2 == 0 }
+                .first { it > 50 }
+        assertEquals(mapIterations, 100)
```

Comment:
For all the assertEquals call, swap the arguments. It should be assertEquals(expected, actual)

Reply:
@KevinGilmore Done